src/bin/main.rs

```diff
@@ -37,22 +39,34 @@ fn main() -> Result<(), Error> {
     ));
 
     // Start web server
-    let (web_send_close, web_close_channel) = tokio::sync::oneshot::channel::<()>();
-    web_rt.spawn(run_web_server(
+    let (web_send_close, web_close_channel) = oneshot::channel::<()>();
+    let (admin_send_close, admin_close_channel) = oneshot::channel::<()>();
+
+    web_rt.spawn(async move {
```

Comment:
I think you're executing a blocking task on an event loop thread here

Reply:
Yes good catch
Maybe I should start this in a separate thread
I think we should take some time after this to think abot our concurrency model and how we want to manage it.

Reply:
I changed for a thread see descritpion